core/src/main/java/com/kin/ecosystem/core/accountmanager/AccountManagerImpl.java

```diff
@@ -96,13 +94,18 @@ public void retry() {
 		}
 	}
 
+	@Override
+	public void logout() {
+		accountState.removeAllObservers();
```

Comment:
also call also call  removeAccountCreationRegistration() before accountState.postValue(REQUIRE_CREATION);  
to eliminate edge cases were logout is called during 

check in removeAccountCreationRegistration()  that AccountCreationRegistration is not null

Reply:
can you add a task (not for now) to better handle switchAccount API?

Reply:
don't understand the task goal, let's talk.